c/datasetjson.c

```diff
@@ -2244,6 +2250,340 @@ void respondWithHLQNames(HttpResponse *response, MetadataQueryCache *metadataQue
+            }
+          }
+          setTextUnitCharOrInt(1, valueString[0], configsCount, DALSYSOU, textUnit);
+        }
+      } else if(!strcmp(propString, "spgnm")) {
+        if (valueStrLen <= CLASS_WRITER_SIZE){
+          setTextUnitString(valueStrLen, &(valueString)[0], configsCount, DALSPGNM, textUnit);
+        }
+      } else if(!strcmp(propString, "close") || !strcmp(propString, "dummy")) {
+        if (!strcmp(valueString, "true")){
+          setTextUnitBool(configsCount, !strcmp(propString, "close") ? DALCLOSE : DALDUMMY, textUnit);
+        }
+      } else if(!strcmp(propString, "dcbdd") || !strcmp(propString, "retdd")) {
+        if (valueStrLen <= DD_NAME_LEN){
+          setTextUnitString(DD_NAME_LEN, &(valueString)[0], configsCount, !strcmp(propString, "dcbdd") ? DALDCBDD : DALRTDDN, textUnit);
+        }
+      } else if(!strcmp(propString, "spin")) {
+        if (!strcmp(valueString, "UNALLOC")){
+          setTextUnitCharOrInt(1, SPIN_UNALLOC, configsCount, DALSPIN, textUnit);
+        } else if (!strcmp(valueString, "ENDJOB")){
+          setTextUnitCharOrInt(1, SPIN_ENDJOB, configsCount, DALSPIN, textUnit);
+        }
+      } else if(!strcmp(propString, "strcls")) {
+        if (valueStrLen <= CLASS_WRITER_SIZE){
+          setTextUnitString(8, &(valueString)[0], configsCount, DALSTCL, textUnit);
+        }
+      } else if(!strcmp(propString, "mngcls") || !strcmp(propString, "datacls")) {
+        if (valueStrLen <= CLASS_WRITER_SIZE){
+         setTextUnitString(CLASS_WRITER_SIZE, &(valueString)[0], configsCount, !strcmp(propString, "mngcls") ? DALMGCL : DALDACL, textUnit);
+        }
+      }
+    }
+    currentProp = jsonObjectGetNextProperty(currentProp);
+  }
+}
+
+static void setTextUnitString(int size, char* data, int *configsCount, int key, DynallocNewTextUnit *textUnit) {
+  textUnit[*configsCount].size = size;
+  textUnit[*configsCount].type = TEXT_UNIT_STRING;
+  textUnit[*configsCount].key = key;
+  textUnit[*configsCount].data.string = data;
+  (*configsCount)++;
+}  
+
+static void setTextUnitCharOrInt(int size, int data, int *configsCount, int key, DynallocNewTextUnit *textUnit) {
+  textUnit[*configsCount].size = size;
+  textUnit[*configsCount].type = TEXT_UNIT_CHARINT;
+  textUnit[*configsCount].key = key;
+  textUnit[*configsCount].data.number = data;
+  (*configsCount)++;
+}  
+
+static void setTextUnitBool(int *configsCount, int key, DynallocNewTextUnit *textUnit) {
+  textUnit[*configsCount].type = TEXT_UNIT_BOOLEAN;
+  textUnit[*configsCount].key = key;
+  (*configsCount)++;
+}
+
+static int getDSCB(char* datasetName, char* dscb, int bufferSize){
+  if (bufferSize < INDEXED_DSCB){
+    zowelog(NULL, LOG_COMP_RESTDATASET, ZOWE_LOG_WARNING, 
+            "DSCB of size %d is too small, must be at least %d", bufferSize, INDEXED_DSCB);
+    return 1;
+  }
+  Volser volser = {0};
+
+  DatasetName dsn = {0};
+  memcpy(dsn.value, datasetName, DATASET_PATH_MAX);
```

Comment:
It would be better to pass `DatasetName *datasetName` as an argument because the code assumes that `datasetName` is blank-padded.